Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1366,75 +1398,67 @@ void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack(GridTrack& track
-                    if (!GridLayoutFunctions::hasRelativeOrIntrinsicSizeForChild(currentAncestorSubgrid, flowAwareDirection))
-                        availableSpace = currentAncestorSubgrid.availableSpaceForGutters(flowAwareDirection);
-
-                    auto gridItemSpanInAncestor = currentAncestorSubgrid.gridSpanForChild(*gridItem, flowAwareDirection);
-                    auto numTracksForCurrentAncestor = currentAncestorSubgrid.numTracks(flowAwareDirection);
-
-                    const auto* currentAncestorSubgridParent = dynamicDowncast<RenderGrid>(currentAncestorSubgrid.parent());
-                    ASSERT(currentAncestorSubgridParent);
-                    if (!currentAncestorSubgridParent)
-                        return std::nullopt;
-
-                    if (gridItemSpanInAncestor.startLine())
-                        gutterTotal += (currentAncestorSubgrid.gridGap(flowAwareDirection) - currentAncestorSubgridParent->gridGap(flowAwareDirection)) / 2;
-                    if (span.endLine() != numTracksForCurrentAncestor)
-                        gutterTotal += (currentAncestorSubgrid.gridGap(flowAwareDirection) - currentAncestorSubgridParent->gridGap(flowAwareDirection)) / 2;
-                    flowAwareDirection = GridLayoutFunctions::flowAwareDirectionForParent(currentAncestorSubgrid, *currentAncestorSubgridParent, flowAwareDirection);
-                }
-                return gutterTotal;
-            }();
             auto accumulatedMbpWithSubgrid = currentAccumulatedMbp + computeSubgridMarginBorderPadding(m_renderGrid, m_direction, track, trackIndex, span, inner.get());
-            track.setBaseSize(std::max(track.baseSize(), accumulatedMbpWithSubgrid + extraMarginFromSubgridAncestorGutters.value_or(0_lu)));
+            track.setBaseSize(std::max(track.baseSize(), accumulatedMbpWithSubgrid + extraMarginFromSubgridAncestorGutters(gridItem, span).value_or(0_lu)));
 
             GridIterator childIterator = GridIterator::createForSubgrid(*inner, iterator);
             accumulateIntrinsicSizesForTrack(track, trackIndex, childIterator, itemsSortedByIncreasingSpan, itemsCrossingFlexibleTracks, itemsSet, accumulatedMbpWithSubgrid);
-            continue;
+            return;
         }
+
         if (!isNewEntry)
-            continue;
-        GridSpan span = m_renderGrid->gridSpanForChild(*gridItem, m_direction);
+            return;
 
         if (spanningItemCrossesFlexibleSizedTracks(span))
             itemsCrossingFlexibleTracks.append(GridItemWithSpan(*gridItem, span));
         else if (span.integerSpan() == 1)
             sizeTrackToFitNonSpanningItem(span, *gridItem, track);
         else
             itemsSortedByIncreasingSpan.append(GridItemWithSpan(*gridItem, span));
-    }
+    };
+
+    while (auto* gridItem = iterator.nextGridItem())
+        accumulateIntrinsicSizes(gridItem);
+
+    for (auto* gridItem : iterator.masonryItems())
+        accumulateIntrinsicSizes(gridItem);
```

Comment:
We probably want to use `CheckedPtr` here

Reply:
Updated to use CheckedPtr.